add env CATTLE_HELM_VERSION

skanakal opened the pull request:
## Issue: <!-- link the issue or issues this PR resolves here --> https://github.com/rancher/rancher/issues/46536
<!-- If your PR depends on changes from another pr link them here and describe why they are needed on your solution section. -->
 
## Problem
<!-- Describe the root cause of the issue you are resolving. This may include what behavior is observed and why it is not desirable. If this is a new feature describe why we need this feature and how it will be used. -->
The CATTLE_HELM_VERSION variable — this was completely [removed](https://github.com/rancher/rancher/pull/47661/commits/356f6c38e9c0a4f5dc111c3c1d193862fe70e585#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caL60) from the backend starting with Rancher v2.11. Since then, Helm version is fetched from the dev environment. However, in upgraded clusters, it still seems to fall back to the old CATTLE_HELM_VERSION value, leading to inconsistencies.
 
## Solution
<!-- Describe what you changed to fix the issue. Relate your changes back to the original issue / feature and explain why this addresses the issue. -->
reintroduce CATTLE_HELM_VERSION
 
## Testing
<!-- Note: Confirm if the repro steps in the GitHub issue are valid, if not, please update the issue with accurate repro steps. -->

## Engineering Testing
### Manual Testing
<!-- Describe what manual testing you did (if no testing was done, explain why). -->
<img width="941" height="533" alt="image" src="https://github.com/user-attachments/assets/6d0a1644-0640-469e-9712-f7e97c23efcf" />


### Automated Testing
<!-- Ensure there are unit/integration/validation tests added (if possible); describe what cases they cover and do not cover. -->
* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  <!-- 
  Non-exhaustive list of reasons:
    - Lack of the framework capable of testing this fix/change
    - Tight deadlines / critical priority to get fix/change in - !ensure GH issue is logged to add tests!
    - No application logic is modified by this change, e.g. refactoring/cosmetic/non-code/test change
    - Tests implemented in another PR elsewhere - !ensure GH PR link is added!
    - Other (explain)
  Note: Outside of the exceptions above, the "existing tests cover the changes" is very unlikely to be an acceptable reason as the existing tests generally don't cover the logic changes implemented by this PR 
  -->
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

## QA Testing Considerations
<!-- Highlight areas or (additional) cases that QA should test w.r.t a fresh install as well as the upgrade scenarios -->
 
### Regressions Considerations
<!-- Dedicated section to specifically call out any areas that with higher chance of regressions caused by this change, include estimation of probability of regressions -->
_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

rohitsakala reviewed (changes requested):
We need to manually set this from `shell` repository every time we update the `helm version`. This is tedious. Can we use some kind of command like this  ?
```
HELM_VERSION=$(curl -s https://raw.githubusercontent.com/rancher/shell/main/hack/make/deps.mk | grep ^HELM_VERSION | sed 's/HELM_VERSION ?= //')
```

Also, can we use it in build.yaml ? https://github.com/rancher/rancher/blob/main/docs/build.md?plain=1

Also, please add a code comment saying this is only used for showcasing the Helm version in the UI.

rohitsakala commented:
> @skanakal It is possible to do it with renovate, but you have to manually configure it. `build.yaml` will be removed soon and replaced with something else.

Renovate is the best choice. I completely forgot about it

skanakal commented:
> @skanakal It is possible to do it with renovate, but you have to manually configure it. `build.yaml` will be removed soon and replaced with something else.

@tashima42 @rohitsakala  If I understand correctly, Renovate will automatically update the versions based on the ENV variable’s currentValue. I’ve opened a PR https://github.com/rancher/renovate-config/pull/501 in renovate-config; please let me know if this is incorrect. I have also adjusted the PR in rancher/rancher accordingly.
